### PR TITLE
Feat - order notifications

### PR DESCRIPTION
**Pull request by alecritson**
Initial work on bringing in order notifications to the hub

---

This PR will also rely on #332

### CODE REVIEW

**Comment by glennjacobs** [comment]
Needs a change log entry and example mailable to show how content is injected.

File: docs/src/getcandy/orders.md [hunk 1]
```diff
@@ -309,3 +309,27 @@ Once updated, GetCandy will keep a render of the email sent out in the activity
...
+Here's an example of what the template could look like:
+
+```html
+<h1>It's on the way!</h1>
+
+<p>Your order with reference {{ $order->reference }} has been dispatched!</p>
+
+<p>{{ $order->total->formatted() }}</p>
+
+@if($content ?? null)
```
Comment (glennjacobs) [inline docs/src/getcandy/orders.md:327]: Is `??` correct here?

Comment (alecritson) [reply docs/src/getcandy/orders.md:327]: Yeah cause if the email is sent via something outside of the order notification, it won't be set potentially.

File: docs/src/getcandy/orders.md [hunk 2]
```diff
@@ -281,6 +281,57 @@ In terms of an order, all it's worried about is whether or not the `placed_at` c
 
 And as always, if you have any questions you can reach out on our Discord!
 
+## Order Notifications
+
+GetCandy allows you to specific what Laravel mailers/notifications should be available for sending when you update an order's status. These are configured in the `getcandy/orders` config file and are defined like so:
```
Comment (glennjacobs) [inline docs/src/getcandy/orders.md:286]: "GetCandy allows you to specific" < should be "specify"?

File: docs/src/getcandy/orders.md [hunk 3]
```diff
@@ -281,6 +281,57 @@ In terms of an order, all it's worried about is whether or not the `placed_at` c
...
+            App\Mail\MyMailer::class,
+            App\Mail\MyOtherMailer::class,
+        ],
+        'notifications' => [],
+    ],
+    // ...
+],
+```
+
+Now when you update an order's status in the hub, you will have these mailers available if the new status is `awaiting-payment`. You can then choose the email addresses which should be sent out and also add an additional email address if required.
```
Comment (glennjacobs) [inline docs/src/getcandy/orders.md:304]: "email addresses which should be sent out" < change to "sent to"?

**Review by glennjacobs — CHANGES_REQUESTED** [review]
Just some doc changes as highlighted, then good to go.

**Review by glennjacobs — APPROVED** [review]

**Pull request merged**